src/vm/diagnosticprotocolhelper.cpp

```diff
@@ -11,17 +11,55 @@
 #ifdef FEATURE_PERFTRACING
 #ifdef FEATURE_PAL
 
-static void WriteStatus(uint64_t result, IpcStream* pStream)
+void DiagnosticProtocolHelper::HandleIpcMessage(DiagnosticsIpc::IpcMessage& message, IpcStream* pStream)
 {
-    uint32_t nBytesWritten = 0;
-    bool fSuccess = pStream->Write(&result, sizeof(result), nBytesWritten);
-    if (fSuccess)
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_ANY;
```

Comment:
Can you make this MODE_PREEMPTIVE?
What makes you think it might have to run in cooperative mode?